pontoon/base/views.py

```diff
@@ -190,10 +200,54 @@ def locale_project(request, locale, slug):
     })
 
 
+def handle_uploaded_file(locale, project, path, f):
+    """
+    Update translations in the database from uploaded file.
+
+    :param Locale locale: Locale instance.
+    :param Project project: Project instance.
+    :param str path: Resource path.
+    :param UploadedFile f: UploadedFile instance.
+    """
+    # Store uploaded file to disk
+    filename = str(uuid.uuid4()) + os.path.splitext(f.name)[1]
+    with codecs.open(filename, 'w') as destination:
+        for chunk in f.chunks():
+            destination.write(chunk)
+
+    # Update database objects from file
+    resource_file = formats.parse(filename)
+    os.remove(filename)
+    entities = {entity.key: entity for entity in Entity.objects.filter(
+        resource__project=project,
+        resource__path=path,
+        obsolete=False
+    )}
+
+    changeset = ChangeSet(
+        project,
+        VCSProject(project, locales=[locale]),
+        timezone.now()
+    )
+    changeset.locales = {locale.code: locale}
+    for vcs_translation in resource_file.translations:
```

Comment:
So this feature is only enabled for translators. And we handle conflicts in pontoon/sync/changeset.py: update_entity_translations_from_vcs.